activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -69,20 +69,36 @@ def new_time(year, mon, mday, hour, min, sec, microsec, offset = nil)
             \z
           /x
 
-          def fast_string_to_time(string)
-            return unless ISO_DATETIME =~ string
-
-            usec = $7.to_i
-            usec_len = $7&.length
-            if usec_len&.< 6
-              usec *= 10**(6 - usec_len)
+          if RUBY_VERSION >= "3.2"
+            def fast_string_to_time(string)
+              return unless ISO_DATETIME.match?(string)
```

Comment:
Actually, isn't there a way we could avoid matching the regexp? How does `Time.new` handle invalid dates?

Reply:
Yes, I tried that, but then found out that the input validation inside `Time.new` was not enough here. For example,
`Time.new('20230123')` returns `20230123-01-01 00:00:00 +0900`-ish Time instance because `Time.new('20230123')` kicks the pre-3.2 initializer.

Please take a look at my report at https://bugs.ruby-lang.org/issues/19293 for more details.

Reply:
Ah I see. That's infortunate. Hopefully we'll have something better in 3.3.